src/colorizer/ColorizeCanvas.ts

```diff
@@ -166,16 +167,17 @@ export default class ColorizeCanvas {
     this.setUniform("highlightedId", id);
   }
 
-  setFeature(name: string): void {
+  getFeatureData(name: string): FeatureData | null {
     if (!this.dataset || !Object.keys(this.dataset.features).includes(name)) {
-      return;
+      return null;
     }
+    return this.dataset.features[name];
+  }
 
-    const { tex, min, max } = this.dataset.features[name];
-
+  setFeature(tex: Texture, min: number, max: number): void {
     this.setUniform("featureData", tex);
     this.setUniform("featureMin", min);
-    this.setUniform("featureMax", max / 1.5);
+    this.setUniform("featureMax", max);
   }
```

Comment:
This API change, and the need to replace `canv.setFeature` with the new function `updateFeature` in main.ts, makes me think that locking behavior should be handled within `ColorizeCanvas`.

The goal of the `ColorizeCanvas` API should be to capture all the actions a user should reasonably want to perform on the canvas, so that main.ts (or any other future client of the class) can primarily focus on setting up UI and attaching it to the appropriate methods. I could see a benefit of splitting up feature switching into getting and setting for error handling purposes, but if implementing locking outside `ColorizeCanvas` requires main.ts to reach into the feature abstraction and manipulate underlying data, that's a sign it should be implemented inside `ColorizeCanvas`.

Reply:
Play/pause is a notable exception, but it gets away with it because it just calls `setFrame` in a smart way. (I'm honestly starting to reconsider keeping it separate though - if we wanted more explicit support, moving the `TimeControls` class into its own file might be a good first step)

Reply:
+1 for moving TimeControls into its own file.  I should have done it when I first added it.

Reply:
@frasercl it took me a while but I think I understand your main point here.  The lock-ability means we have to provide max and min independently of the feature data texture.  It does seem to make sense that ColorizeCanvas could be responsible for holding on to the locked min/max and the flag for whether to use the "locked" min/max or the Feature's own min/max.
What's the api then?
ColorizeCanvas.setMinMax(number, number) or setLockedMinMax(number, number)?  ColorizeCanvas.setLockMinMax(bool) ?

Reply:
I think `setLockMinMax` and `setMinMax` (`setFeatureRange`/`setLockFeatureRange`?) seem reasonable, though Peyton pointed out that getters for the min/max will be more important than setters at this stage. The UI needs access to the min/max values to display now, but `ColorizeCanvas` can set the min/max off of the feature until we build in the option to set the values manually later.

Reply:
Talked to Peyton and realized you don't need min/max setters, as the internals can just grab the values when lock is set to true.

Reply:
@toloudis I agree with what @frasercl said after we had a chance to talk about it in person-- it makes sense that the logic in main.ts should be a UI wrapper around ColorizeCanvas, rather than storing and updating the range values itself.

I added getters for the min and max color map ranges and a setter for the lock (`setColorMapRangeLock(bool)`) into ColorizeCanvas, which are accessed from main.ts. I'll eventually need to add setters for the range values too, but that's part of a different feature that will come later.